dpctl/tensor/libtensor/source/reduction_over_axis.hpp

```diff
@@ -79,15 +80,24 @@ inline bool check_atomic_support(const sycl::queue &exec_q,
     return supports_atomics;
 }
 
+template <bool return_value>
+bool fixed_decision(const sycl::queue &, sycl::usm::alloc)
```

Comment:
With this, it will probably be possible to refactor ``argmin`` / ``argmax`` to use the same ``py_reduction_over_axis`` binding

Reply:
This is a dynamic check. Things are likely to fall apart during compilation as compiler will still need to generate code for atomic implementation. Worth a try though.